### ISSUE TITLE
Merging QCQP and Conic Formulations

### ISSUE THREAD

**Issue by ccoffrin** [lanl-ansi/PowerModels.jl#436]
Based on discussion with @mlubin and @blegat, the best longterm strategy is to implement all formulations in PowerModels using MOI's SOC and RSOC constraints. Bridges than can be used to send those JuMP models to QCQP or NLP solvers.  These bridges may be implemented in InfrastructureModels or as optional bridges provided as part of MOI.

To that end, cleaning up the conic PowerModel types will wait until PowerModels moves to JuMP v0.19.

CC @frederikgeth, @jd-lara

**Comment by jd-lara** [comment]
@ccoffrin but it is still feasible to clean up the upper level of the type tree to implement it in PowerSimulations?

**Comment by ccoffrin** [comment]
Yes, the active power only type is independent and will get into the next release of PowerModels.

**Comment by frederikgeth** [comment]
That's great. Excited about all the features JuMP 0.19 is going to offer!

**Comment by ccoffrin** [comment]
@blegat and @odow, the ecosystem has come a long way since this issue was originally posted.  What is the current best practice on this topic?

At least in the near term, a good implementation of quadratic objective functions in the conic solvers that only support linear objectives will keep these two types around, but I would be nice if there was a path to remove these in the future.

**Comment by odow** [comment]
All solvers have a quadratic to SOC bridge by default, but the SOC to quadratic is opt-in. So you can use the QCQP formulations with all solvers.

Ideally, you could just keep the conic formulations. But you'd probably need to add the `SOCtoNonConvexQuadBridge{Float64}` bridge to your JuMP model.

@blegat can say more.

**Comment by blegat** [comment]
For constraints `norm(U*x) <= t`, use SOC. Even for QCQP solvers, its best to do `y = U*x`, `sum(y.^2) <= t^2` than to to do `xU'Ux <= t^2`
